### ISSUE TITLE
JupyterLab-Manager messed up Jupyter Lab layout

### ISSUE THREAD

**Issue by ivan-monge1989** [jupyter-widgets/ipywidgets#2545]
Hi,

I tried installing Lab Extensions by running the following codes in the terminal in my Macbook.
conda install -c conda-forge ipywidgets
conda install -c conda-forge nodejs

up to this point eveything was going all right (except no widgets displayed) but after I ran 
jupyter labextension install @jupyter-widgets/jupyterlab-manager

My JupyterLab layout got super messed up and I cant access any of the side bars.
<img width="1440" alt="Screen Shot 2019-08-29 at 3 09 30 AM" src="https://user-images.githubusercontent.com/21143059/63918888-9ed94180-ca0b-11e9-8ebe-414e6f90cfb6.png">
<img width="1440" alt="Screen Shot 2019-08-29 at 3 09 16 AM" src="https://user-images.githubusercontent.com/21143059/63918891-a13b9b80-ca0b-11e9-8bfa-87015bfbeb45.png">

I've tried uninstalling and reinstalling the manager but, no success. I must point out that the widgets now work.

Any help will be much appreciated.

**Comment by grzegorz700** [comment]
Hi, I had the same bug as you. I had `jupyterlab 1.0.2` with this bug. I have fixed it by doning several steps.

- Uninstall the previous version of jupyterlab-manager  by `jupyter labextension uninstall @jupyter-widgets/jupyterlab-manager` command.
- Clean trash by `jupyter lab clean`
- Upgrade jupyterlab to `jupyterlab 1.1.0`.
- Install `jupyterlab-manager` by extension manager.
- I got a message about rebuild ("A build is needed to include the last changes"), so I clicked `Rebuild`. After that, I got confirmation "Build complete".

**Comment by ivan-monge1989** [comment]
> Hi, I had the same bug as you. I had `jupyterlab 1.0.2` with this bug. I have fixed it by doning several steps.
> 
> * Uninstall the previous version of jupyterlab-manager  by `jupyter labextension uninstall @jupyter-widgets/jupyterlab-manager` command.
> * Clean trash by `jupyter lab clean`
> * Upgrade jupyterlab to `jupyterlab 1.1.0`.
> * Install `jupyterlab-manager` by extension manager.
> * I got a message about rebuild
> 
> > "A build is needed to include the last changes"
> 
> So I clicked `Rebuild`. After that, I got confirmation "Build complete".

How would I do this? I am using Jupyter Lab from the Anaconda Package.

**Comment by LukaPitamic** [comment]
I have the same problem, with @grzegorz700 protocol I've been able to sort it out. But Now I'm getting conflicting dependencies, because current version of Plotly supports JupyterLab v1.0, not v1.1

**Comment by jasongrout** [comment]
> But Now I'm getting conflicting dependencies, because current version of Plotly supports JupyterLab v1.0, not v1.1

Can you be more specific about exactly what dependencies are conflicting?

**Comment by LukaPitamic** [comment]
sure:

```
"jupyterlab-chart-editor@1.2.0" is not compatible with the current JupyterLab
Conflicting Dependencies:
JupyterLabExtension        Package
>=16.8.4 <16.9.0>=16.4.0 <16.5.0 react
>=16.8.4 <16.9.0>=16.4.0 <16.5.0 react-dom
```

**Comment by jasongrout** [comment]
JLab 1.0 depended on react `~16.8.4` (https://github.com/jupyterlab/jupyterlab/blob/e2fd4c8841a7393f9131fb6b6e8252864a8bd351/dev_mode/package.json#L84), which conflicts with jupyterlab-chart-editor's dependency on `~16.4.0` (https://github.com/plotly/jupyterlab-chart-editor/blob/9035bb6241538140b808b065be23cd73714f942e/package.json#L40). Did it really work in 1.0.0? Was react not required to be a singleton in 1.0.0?

**Comment by LukaPitamic** [comment]
It was working for sure in JupyterLab v0.35
I want to re-test it on v1.0 following [official Plotly installation gude](https://plot.ly/python/getting-started/), but I cannot get passed this:

![image](https://user-images.githubusercontent.com/34914144/64246998-723a9380-cefd-11e9-8696-8e02500409af.png)

...as soon as I install `@jupyter-widgets/jupyterlab-manager` over JupyterLab v1.0 I'm stuck with broken left sidebar. 

JupyterLab v1.1 does solve left sidebar, but I cannot get JupyterLab v1.0 and jupyterlab-manager v1.0 to work together without broken left sidebar. ...as suggested by Plotly, which eliminates the above mentioned conflicting dependencies.

Strange, because it was working a week or so ago.

**Comment by LukaPitamic** [comment]
On macOX(OSX) I solved the problem:

It appears this is issue with JupyterLab v 1.0.1, so first check the version you have installed:
`jupyter lab -V`

Then either downgrade from JupyterLab v 1.0.2 to v 1.0.1:
```
conda install -c conda-forge jupyterlab=1.0.1
jupyter lab clean
jupyter lab build
```

...or upgrade to v1.1
```
conda install -c conda-forge jupyterlab=1.1
jupyter lab clean
jupyter lab build
```

**Comment by jasongrout** [comment]
Definitely use 1.1.1 instead of 1.1 :)

**Comment by jasongrout** [comment]
> Right now Plotly does not officially support JupyterLab v1.1

It should work fine in JupyterLab 1.1. There is a known issue of widgets not working if you open a notebook, then close it, then open it again which we're working through, but my guess is that would be the only problem.

**Comment by ivan-monge1989** [comment]
I kinda gave up. I will just wait when Conda officially has 1.1.1 on it's official release. Right now it just has 1.0.2. I will continue using Jupyter Notebook for widgets until this is solved.

**Comment by LukaPitamic** [comment]
@jasongrout I agree, after some testing back and forward I ended up having it all working on 1.1.1. I also updated my comment above as only upgrading JLab is to be viable option to solve the issue. 

The only issue I'm facing is incompatibilities with extension `jupyterlab-chart-editor@1.2.0` from Plotly, which I might live a month or so without. Until Plotly guys update the version to be fully compatible.

@ivan-monge1989 If you still haven't given up really - I recommend you create file `~/.condarc` if you don't have it already. Then add lines:
```
channels:
  - defaults
  - conda-forge
channel_priority: true
```
This will always check for version in default channel, and if it is not there check alternative channel `conda-forge`. This means the line:
`# conda install jupyterlab=1.1`
 ...will install JLab today from `conda-forge` and same line will install in a week or so from `deafault`. I hope this helps.

**Comment by TheCedarPrince** [comment]
@LukaPitamic - I can confirm that your fix worked for me and I am using Ubuntu 16.04. 

I had issues with conda's my version - turns out I was using Jupyter Lab version 1.0.6. Once I upgraded, cleaned, and built, everything worked great and I was able to successfully access my tabs and side bar again. 

Thanks!

**Comment by dstran** [comment]
I can confirm jupyterlab 1.1.2 also worked for me to fix the top and left side layout problems.

**Comment by jasongrout** [comment]
The root of the problem here was that we had some accidental incompatibilities in our 1.1 set of packages, and those were being pulled into the 1.0.x builds. A few days ago we released new versions of packages that were backwards compatible, so the problem should be resolved for 1.0.x now if you clean and rebuild `jupyter lab clean` `jupyter lab build`.

**Comment by jasongrout** [comment]
Fixed by https://github.com/jupyterlab/jupyterlab/pull/7182
